Lib/asyncio/base_events.py

```diff
@@ -647,6 +651,68 @@ def _getaddrinfo_debug(self, host, port, family, type, proto, flags):
         return await self.run_in_executor(
             None, socket.getnameinfo, sockaddr, flags)
 
+    async def sock_sendfile(self, sock, file, offset=0, count=None,
+                            *, fallback=True):
+        if self._debug and sock.gettimeout() != 0:
+            raise ValueError("the socket must be non-blocking")
+        self._check_sendfile_params(sock, file, offset, count)
+        try:
+            return await self._sock_sendfile_native(sock, file,
+                                                    offset, count)
+        except _SendfileNotAvailable:
+            if fallback:
+                return await self._sock_sendfile_fallback(sock, file,
+                                                          offset, count)
+            else:
+                raise
```

Comment:
`raise RuntimeError('os.sendfile is not available') from None`

Reply:
Proposed message text is not always correct: sometimes `os.sendfile` is available but passed file object cannot be used by syscall.
`_SendfileNotAvailable` is derived from `RuntimeError` now and it always contains proper message text.
The best what I can do is reraising:
`raise RuntimeError(exc.args[0]) from None`

Reply:
In this case let's make the exception public. I'd rename it to `SendfileNotAvailableError` and export it to asyncio module. Don't wrap it then.